docs/builder.py

```diff
@@ -0,0 +1,39 @@
+from functools import cached_property
+
+from sphinxcontrib.serializinghtml import JSONHTMLBuilder
+
+
+class PythonObjectsJSONHTMLBuilder(JSONHTMLBuilder):
+    name = "pyjson"
+
+    @cached_property
+    def domain_objects(self):
+        domain = self.env.get_domain("py")
+        return [item for item in domain.get_objects() if item[2] != "module"]
```

Comment:
It would be a bit more readable if we could wrap the items returned by `get_objects()` into something like a dataclass:
```python
from dataclasses import dataclass

@dataclass
class DomainObject:
    """
    A wrapper around sphinx's domain object descriptions
    https://www.sphinx-doc.org/en/master/extdev/domainapi.html#sphinx.domains.Domain.get_objects
    """
    name: str
    dispname: str
    type: str
    docname: str
    anchor: str
    priority: int
```

Oh, and also I'd create either a class or module-level constant `IGNORED_DOMAIN_TYPES = {"module"}` to make it a bit clearer why `"module"` is left out.

Then this all becomes:
```suggestion
        return [item for item in map(DomainObject, domain.get_objects() if item.type not in IGNORED_DOMAIN_TYPES]
```

Reply:
When I applied this exactly, I got this error:
```
    for item in map(DomainObject, domain.get_objects())
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: DomainObject.__init__() missing 5 required positional arguments: 'dispname', 'type', 'docname', 'anchor', and 'priority'
```
I have applied something similar